tests/darcy_velocity.prm

```diff
@@ -0,0 +1,147 @@
+set Start time                             = 0
+set End time                               = 5000
+set Use years in output instead of seconds = true
+set Nonlinear solver scheme                = iterated Advection and Stokes
+set Max nonlinear iterations               = 5
+set CFL number                             = 0.2
+set Output directory                       = darcy_velocity
+set Pressure normalization                 = surface
+set Surface pressure                       = 0
+set Maximum time step                      = 1000
+
+subsection Geometry model
+  set Model name = box
+
+  subsection Box
+    set X extent = 200000
+    set Y extent = 100000
+    set X repetitions = 2
+  end
+end
+
+# This model is only designed to predict fluid velocities arising from initial prescribed values in porosity.
+# As such, no variations in temperature are needed and both the boundary and initial temperatures are held constant.
+
+subsection Initial temperature model
+  set Model name = function
+  subsection Function
+    set Function expression       = 293
+  end
+end
+
+subsection Boundary temperature model
+  set List of model names = box
+  set Fixed temperature boundary indicators   = top, bottom, left, right
+  subsection Box
+    set Bottom temperature = 293
+    set Top temperature    = 293
+    set Left temperature   = 293
+    set Right temperature  = 293
+  end
+end
+
+subsection Boundary velocity model
+  set Prescribed velocity boundary indicators = left x:function, right x:function
+  set Tangential velocity boundary indicators = top, bottom
+  subsection Function
+    set Coordinate system = cartesian
+    set Variable names = x,y,t
+    set Function expression = 1;0
+  end
+end
+
+############### Compositional fields
+# We want to use three compositional fields, the porosity and
+# and two additional fields. The solid field will be advected with
+# the solid speed prescribed above, and the fluid field will be advected
+# as the sum of the solid velocity and the vertical buoyancy the fluid has.
+# The porosity field is uniform to allow for a uniform fluid velocity. The 
+# fluid field is a circle of radius 12.5km at the center of the model domain.
+subsection Compositional fields
+  set Number of fields = 3
+  set Names of fields = porosity, solid, fluid
+  set Compositional field methods = darcy field, field, darcy field
+  set Types of fields = porosity, generic, generic
+end
+
+subsection Initial composition model
+  set Model name = function
+  subsection Function
+    set Variable names      = x,y
+    set Function constants  = pi=3.1415926,x0=100000,y0=50000,c=12500, A_f=0.2, A_s = 1.0
+    set Function expression = A_f; if((x - x0)*(x - x0) + (y - y0)*(y - y0) <= c*c, A_s, 0); if((x - x0)*(x - x0) + (y - y0)*(y - y0) <= c*c, A_s, 0)
+  end
+end
+
+subsection Boundary composition model
+  set List of model names = initial composition
+  set Fixed composition boundary indicators = bottom, top, left, right
```

Comment:
Does this actually work correctly?
I thought you would need to `set Allow fixed composition on outflow boundaries = true` fot the boundary conditions to be applied properly.

Reply:
it seemed like it was working but I added this for good measure